### PR TITLE
Reset error state on switching tab and reset form state

### LINKED ISSUE #576: Error bar sometimes have no content
1. SSO -> enable any one of them and press save without further configuration
Screenshot 2020-10-23 at 14 51 17

2. Try to save an incorrect configuration on Authentication > Authenticator
Screenshot 2020-10-23 at 14 52 18

### PR DESCRIPTION
**Pull request by axolotl0212**
fix #576 #549

### CODE REVIEW

File: portal/src/hook/usePivot.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ function isHashValid(validItemKeys: string[], hash: string): boolean {
 }
 
 export function usePivotNavigation(
-  validItemKeys: string[]
+  validItemKeys: string[],
+  switchTabEffect?: () => void
```
Comment (louischan-oursky) [inline portal/src/hook/usePivot.ts:11]: It is better to call this `onSwitchTab`

File: portal/src/graphql/adminapi/AddEmailScreen.tsx [hunk 2]
```diff
@@ -63,8 +64,9 @@ const AddEmailScreen: React.FC = function AddEmailScreen() {
   }, [initialFormData, formData]);
 
   const resetForm = useCallback(() => {
+    resetCreateIdentityError();
     setFormData(initialFormData);
-  }, [initialFormData]);
+  }, [initialFormData, resetCreateIdentityError]);
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/AddEmailScreen.tsx:69]: Can we use `key` to force re-mount of the component to implement form reset?

Comment (nicolaslee2775) [reply portal/src/graphql/adminapi/AddEmailScreen.tsx:69]: Currently, only `PasswordlessAuthenticatorScreen` and `PasswordsScreen` use key to force re-mount components. Maybe create new issue for updating all other screens?

Comment (louischan-oursky) [reply portal/src/graphql/adminapi/AddEmailScreen.tsx:69]: I prefer using key to re-mount the form to reset the form. It is simpler and less error-prone. Would a new abstraction (I don't know, maybe a hook) be helpful here? Maybe you guys can discuss and come up with a solution.

Comment (axolotl0212) [reply portal/src/graphql/adminapi/AddEmailScreen.tsx:69]: For some of the screen we can use key to re-mount and reset the form, but for some other screen, the mutation hook is called on screen level, so remounting the form does not reset the error state (for example screen with tab). Then we will need to change the component structure (say introduce another wrapper element, call mutation in its child and remount its child on reset).

Comment (nicolaslee2775) [reply portal/src/graphql/adminapi/AddEmailScreen.tsx:69]: Use can keep `resetCreateIdentityError()` and change `setFormData(initialFormData)` to `setRemountIdentifier((prev) => prev + 1)`.

**Comment by axolotl0212** [comment]
@louischan-oursky @nicolaslee2775 updated and rebased. For some screen we will use resetError as @nicolaslee2775 suggested, and use key to remount form for all screen to reset form state

**Pull request merged**